In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from statsmodels.tsa.api import SimpleExpSmoothing, Holt, ExponentialSmoothing
from sklearn.metrics import mean_squared_error

In [2]:
df = pd.read_excel('CocaCola_Sales_Rawdata.xlsx')

In [3]:
df

Quarter        Sales
0    Q1_86  1734.827000
1    Q2_86  2244.960999
2    Q3_86  2533.804993
3    Q4_86  2154.962997
4    Q1_87  1547.818996
5    Q2_87  2104.411995
6    Q3_87  2014.362999
7    Q4_87  1991.746998
8    Q1_88  1869.049999
9    Q2_88  2313.631996
10   Q3_88  2128.320000
11   Q4_88  2026.828999
12   Q1_89  1910.603996
13   Q2_89  2331.164993
14   Q3_89  2206.549995
15   Q4_89  2173.967995
16   Q1_90  2148.278000
17   Q2_90  2739.307999
18   Q3_90  2792.753998
19   Q4_90  2556.009995
20   Q1_91  2480.973999
21   Q2_91  3039.522995
22   Q3_91  3172.115997
23   Q4_91  2879.000999
24   Q1_92  2772.000000
25   Q2_92  3550.000000
26   Q3_92  3508.000000
27   Q4_92  3243.859993
28   Q1_93  3056.000000
29   Q2_93  3899.000000
30   Q3_93  3629.000000
31   Q4_93  3373.000000
32   Q1_94  3352.000000
33   Q2_94  4342.000000
34   Q3_94  4461.000000
35   Q4_94  4017.000000
36   Q1_95  3854.000000
37   Q2_95  4936.000000
38   Q3_95  4895.000000
39   Q4_95  4333.000000
40   Q1_96  4194.000000
41   Q2_96  5253.000000

In [4]:
train_data = df.head(30)
test_data = df.tail(12)

In [5]:
ses_model = SimpleExpSmoothing(train_data['Sales']).fit(optimized=True)
ses_pred = ses_model.forecast(len(test_data))
ses_rmse = np.sqrt(mean_squared_error(test_data['Sales'], ses_pred))
print('RMSE value of Simple Exponential Smoothing:', ses_rmse)

RMSE value of Simple Exponential Smoothing: 860.8833554735047


In [6]:
holt_model = Holt(train_data['Sales']).fit(optimized=True)
holt_pred = holt_model.forecast(len(test_data))
holt_rmse = np.sqrt(mean_squared_error(test_data['Sales'], holt_pred))
print('RMSE value of Holt\'s Linear Trend Model:', holt_rmse)

RMSE value of Holt's Linear Trend Model: 518.1409301044574


In [7]:
hw_model = ExponentialSmoothing(train_data['Sales'], seasonal_periods=4, trend='add', seasonal='add').fit()
hw_pred = hw_model.forecast(len(test_data))
hw_rmse = np.sqrt(mean_squared_error(test_data['Sales'], hw_pred))
print('RMSE value of Holt-Winter\'s Model:', hw_rmse)

RMSE value of Holt-Winter's Model: 231.9058628876914


Based on the RMSE values, Holt-Winter's Model has the lowest RMSE value of 231.91. 
Therefore, we will use Holt-Winter's Model for forecasting the CocaCola dataset.